### PR TITLE
fabrics: Update filter functions and add local ana state getter

### PR DESCRIPTION
**Pull request by igaw**
The filters API has changed in libnvme update the callsides in
nvme-cli. Also implement the nvme_ctrl_get_ana_state() in nvme-cli as
the library has now all APIs to retrieve this information.

Signed-off-by: Daniel Wagner <dwagner@suse.de>

### CODE REVIEW

File: nvme-print.c [hunk 1]
```diff
@@ -2373,6 +2373,24 @@ void nvme_show_supported_cap_config_log(
...
+	nvme_ctrl_for_each_path(c, p) {
+		n = nvme_path_get_ns(p);
+		if (!n || nvme_ns_get_nsid(n) != nsid)
+			continue;
+		return nvme_path_get_ana_state(p);
+	}
+
+	return NULL;
+}
+
```
Comment (hreinecke) [inline nvme-print.c:2393]: Errm. But this is precisely why I _did_ do the rework.
The filter function is supposed to prune the tree from all unrelated entries, and we just have to print the tree itself.
Let me push my changes here.

**Review by hreinecke — CHANGES_REQUESTED** [review]

**Comment by hreinecke** [comment]
Check PR #1482.

File: nvme-print.c (hunk 1 above)
Comment (igaw) [reply nvme-print.c:2393]: The pruning works too good in my case...

**Comment by igaw** [comment]
Got replaced by #1485 (which is base on #1482)

**Pull request closed without merging**